src/pepr/operator/controllers/exemptions/exemptions.ts

```diff
@@ -46,117 +34,127 @@ function addIfIncludesPolicy(
 function deleteIfPolicyRemoved(
   policy: Policy,
   policyMap: PolicyMap,
-  matcherPolicies: Policy[],
-  matcherName: string,
+  exemptionEl: ExemptionElement,
   ownerId: string,
-  storedMatchers: StoredMatcher[],
 ) {
-  if (storedMatchers.some(sm => sm.name === matcherName) && !matcherPolicies.includes(policy)) {
+  const matcher = {
+    ...exemptionEl.matcher,
+    owner: ownerId,
+  };
+  const storedMatchers = policyMap.get(policy) ?? [];
+
+  if (storedMatchers.some(sm => isSame(sm, matcher)) && !exemptionEl.policies.includes(policy)) {
     policyMap.set(
       policy,
       storedMatchers.filter(sm => {
-        if (sm.name !== matcherName || (sm.name === matcherName && sm.owner !== ownerId)) return sm;
+        if (!isSame(sm, matcher)) return sm;
       }),
     );
-    Log.debug(`Removing ${matcherName} from ${policy}`);
+    Log.debug(`Removing ${matcher.name} from ${policy}`);
   }
 }
 
 // Delete matchers from the store if they no longer exist on a UDSExemption
 function deleteIfMatchersRemoved(
-  policyMap: PolicyMap,
   policy: Policy,
-  currExemptMatchers: Set<string>,
+  policyMap: PolicyMap,
+  currExemptMatchers: StoredMatcher[],
   ownerId: string,
 ) {
   const policyMatchers = policyMap.get(policy) || [];
 
-  for (const m of policyMatchers) {
-    if (m.owner === ownerId && !currExemptMatchers.has(m.name)) {
+  // Check stored matchers that have same owner ref as current UDSExemption
+  for (const pm of policyMatchers.filter(m => m.owner === ownerId)) {
+    let shouldBeRemoved = true;
+
+    // check if stored matcher exists in current list of UDSExemption matchers
+    for (const m of currExemptMatchers) {
+      if (isSame(pm, m)) {
+        shouldBeRemoved = false;
+      }
+    }
```

Comment:
So I'm not sure this handles all the issues. Here's a minimal reproduction of the issue I'm seeing:
```yaml
apiVersion: uds.dev/v1alpha1
kind: Exemption
metadata:
  name: test
  namespace: uds-policy-exemptions
spec:
  exemptions:
    - policies:
        - DisallowHostNamespaces
      matcher:
        namespace: test
        name: "^test-pod.*"
    - policies:
        - DisallowPrivileged
      matcher:
        namespace: test
        name: "^test-pod.*"
    - policies:
        - RequireNonRootUser
      matcher:
        namespace: test
        name: "^test-pod.*"
    - policies:
        - DropAllCapabilities
      matcher:
        namespace: test
        name: "^test-pod.*"
```

If you apply this manifest the peprstore ends up only having the `DropAllCapabilities` exemption. This would be true anytime someone uses the same exact matcher in the same CR (i.e. anytime name, namespace, kind, and owner line up). I do think is a quite viable situation that we should support if we can - someone might want to separate out exemptions within the same CR so that they could put specific descriptions on each for example. Happy to help brainstorm how to handle this.

Reply:
Extra context for why this might be useful - currently the promtail/neuvector exemptions share a single description for a number of policies that are exempted. In reality those descriptions only explain why one or two of the policies are exempted, not all of them. If we had an easy way to add a description per policy I think we'd be able to provide better info here (that is eventually pulled in UDS Engine?).

Reply:
the mode for giving justifications is not the best atm for sure. I think our options right now are:

1. make policies objects
```yaml
    - policies:
        - name: DropAllCapabilities
          reason: "cause i want to"
        - name: DropAllCapabilities
          reason: "cause i want to"
      matcher:
        namespace: neuvector
        name: "^neuvector-prometheus-exporter-pod.*"
      title: "neuvector-prometheus-exporter-pod"
```
pros:
* nicely attaches a justification to each policy for this matcher, so easier for reporting purposes.
* wouldn't require any controller logic changes
cons:
* doesn't actually fix the above issue. it does give less reason to structure an Exemption that way though.
* could make the exemption blocks messy and less readable if there are big justifications mixed into the policy list.
* have to update CRD / CR, which is not too big a deal since uds-core isn't widely used but is a consideration.

2. Squish
Like you mentioned in chat, we could take all the duplicate matchers and combine them into a single exemption block at time of processing.

pros:
* no CRD / CR updates, just controller logic changes.
* handles the actual bug you found.
cons:
* might or might not solve the justification clarity issue and doesn't promote our pattern

3. Make titles unique and required
If we make titles unique and required, we can do comparisons based of those and not just kind, ns, name, and owner. 

pros:
* users have to give titles, which helps with reporting
* we don't have to generate titles for them if they're missing
* would be a fix for the bug
cons:
* might just make controller logic more complicated not less.
* CRD / CR has to change a little bit
* might or might not solve the justification clarity issue and doesn't promote our pattern

4. Do an additional check
Since `deleteIfPolicyRemoved()` is the culprit here, we could probably do a check for any duplicate matcher that does have the policy before deleting the matcher off the policy. 

pros:
* no CRD / CR changes needed
* handles bug
cons:
* could add a performance hit (probably negligible though)
* probably make the code more bleh
* might or might not solve the justification clarity issue and doesn't promote our pattern

I think i'm liking option 1 and / or option 2.

Reply:
Opened an issue for this, I think we can resolve in a follow on.